### PR TITLE
Performance Boost and Code Cleanup

### PR DESCRIPTION
**Pull request by ericlancebrown**

### CODE REVIEW

**Comment by nekiro** [comment]
Describe your changes please.

**Comment by ericlancebrown** [comment]
> Describe your changes please.

Will update asap with a description.

I'm moving today so I am not able.

**Comment by ericlancebrown** [comment]
Changes:

I had included the const std::string& instead of std::string in many of the vars.

Passing as a const reference instead of a value is faster and recommended.

Cleaned up some unnecessary code. Just unused functions and variables that were giving me warnings to fix those.

Nothing too drastic but hopefully will increase performance and readability of the code.

**Comment by ericlancebrown** [comment]
meme

File: src/iologindata.cpp [hunk 1]
```diff
@@ -44,13 +44,6 @@ Account IOLoginData::loadAccount(uint32_t accno)
 	return account;
 }
 
-bool IOLoginData::saveAccount(const Account& acc)
```
Comment (nekiro) [inline src/iologindata.cpp:47]: why are you removing it?

File: src/protocolgame.h [hunk 2]
```diff
@@ -43,11 +43,11 @@ struct TextMessage
 	MessageClasses type = MESSAGE_STATUS_DEFAULT;
 	std::string text;
 	Position position;
-	uint16_t channelId;
+	uint16_t channelId{};
```
Comment (nekiro) [inline src/protocolgame.h:46]: ?

File: src/script.cpp [hunk 3]
```diff
@@ -37,7 +37,7 @@ Scripts::~Scripts()
 	scriptInterface.reInitState();
 }
 
-bool Scripts::loadScripts(std::string folderName, bool isLib, bool reload)
+bool Scripts::loadScripts(const std::string folderName, bool isLib, bool reload)
```
Comment (nekiro) [inline src/script.cpp:40]: did you miss &?

File: src/spells.cpp [hunk 4]
```diff
@@ -264,16 +264,6 @@ InstantSpell* Spells::getInstantSpell(const std::string& words)
 	return nullptr;
 }
 
-InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
```
Comment (nekiro) [inline src/spells.cpp:267]: why removing it?

File: src/protocolgame.h [hunk 5]
```diff
@@ -43,11 +43,11 @@ struct TextMessage
...
 	struct {
 		int32_t value = 0;
 		TextColor_t color;
-	} primary, secondary;
+	} primary{}, secondary{};
```
Comment (ranisalt) [inline src/protocolgame.h:50]: Not needed.

File: src/iomarket.cpp [hunk 6]
```diff
@@ -50,6 +50,7 @@ MarketOfferList IOMarket::getActiveOffers(MarketAction_t action, uint16_t itemId
 		offer.price = result->getNumber<uint32_t>("price");
 		offer.timestamp = result->getNumber<uint32_t>("created") + marketOfferDuration;
 		offer.counter = result->getNumber<uint32_t>("id") & 0xFFFF;
+		offer.itemId = result->getNumber<uint16_t>("itemtype");
```
Comment (ranisalt) [inline src/iomarket.cpp:53]: Unrelated?

File: src/iologindata.cpp (hunk 1 above)
Comment (ranisalt) [reply src/iologindata.cpp:47]: Probably, it's not used in the code, both this and the other function he removed.

**Review by ranisalt — CHANGES_REQUESTED** [review]

**Comment by ranisalt** [comment]
> Passing as a const reference instead of a value is faster and recommended.

Note that passing a value instead of a reference does not necessarily imply in an extra copy. Compilers are smart and avoid copying as much as possible, so if you have a value that is not changed in the function body, it most likely generates code for a reference or move anyway.

To have improvements, you should use perfect forwarding (also called universal reference), [see this post](https://stackoverflow.com/a/17644402).

**Comment by DSpeichert** [comment]
@ericlancebrown do you intend to update this PR per reviews?

**Comment by ericlancebrown** [comment]
Nah I have gotten too busy. Sorry.
